### ISSUE TITLE
VideoCLIP: Contrastive Pre-training for Zero-shot Video-Text Understanding

### ISSUE THREAD

**Issue by hypknot74** [hypknot74/Paper-summaries#59]
cvpaper.challengeのまとめの補足だけ書いている

### リンク

paper: https://aclanthology.org/2021.emnlp-main.544/

### 他参考サイト
* cvpaper.challengeのまとめ
![image](https://github.com/hypknot74/Paper-summaries/assets/57225520/b037c85f-838f-4a34-b5b6-2540d15a3a3e)

* [文献紹介：VideoCLIP: Contrastive Pre-training for Zero-shot Video-Text Understanding](https://www.slideshare.net/ttamaki/videoclip-contrastive-pretraining-for-zeroshot-videotext-understanding)

### 著者/所属機関
Hu Xu, Gargi Ghosh, Po-Yao Huang, Dmytro Okhonko, Armen Aghajanyan, Florian Metze, Luke Zettlemoyer, Christoph Feichtenhofer
Facebook AI, Carnegie Mellon University

### 投稿日付
EMNLP 2021

### Contrastive Loss
video-textのcontrastive lossを採用している。

![image](https://github.com/hypknot74/Paper-summaries/assets/57225520/2483b77e-1d04-4a64-b71b-24bd640c90c9)

NCEは具体的には[InfoNCE](https://qiita.com/omiita/items/a7429ec42e4eef4b6a4d#12-%E3%82%A8%E3%83%B3%E3%82%B3%E3%83%BC%E3%83%80%E3%83%BC%E3%81%AE%E5%AD%A6%E7%BF%92)を使用。

![image](https://github.com/hypknot74/Paper-summaries/assets/57225520/ec08c92a-90e2-41dc-b944-1965e3829ddc)

### Overlapped Video-Text Clips
video-text事前学習では、時間的に完全に整列したクリップ（動画とテキストのクリップの開始/終了タイムスタンプが同じ）であることが一般的だが、そのまま使用するのは良くない。

例えば、「人が話している」動画クリップとタイムスタンプを共有するテキストクリップは「私はあなたにチャーハンの作り方を教えます」かもしれないが、これらの関連性は低い。むしろ後の「フライパンにごはんを入れる」動画クリップのほうが関連性が高い。人間は同時に話すことと行動することが少ないため、動画クリップとテキストクリップでは時間的に完全にアラインメントを取ると、関連性が低くなる。

よって、下記の3ステップでVideo-Text Clipsを作成する。

1. テキストクリップをサンプリングする（最初にビデオクリップをサンプリングすると、対応するテキストが近くにない可能性があるため）。
2. テキストクリップの範囲内でランダムなタイムスタンプをサンプリングし、ビデオクリップの中心とします。
これは、選択したテキストクリップがビデオのどの部分に対応するかを決定するためのステップである。テキストクリップの時間範囲内からランダムに一つの時間点（タイムスタンプ）を選択し、そのタイムスタンプをビデオクリップの中心とする。
3. このタイムスタンプからランダムな期間（最大約32秒）をビデオクリップとする。

これらの手順により、時間的に一部が重なるが、必ずしも完全に一致しないビデオクリップとテキストクリップのペアを作成できる。

### Retrieval Augmented Training
↓上の参考資料から引用
![image](https://github.com/hypknot74/Paper-summaries/assets/57225520/853e2d54-6392-4aef-9197-c4ecc526b1c6)
これはつまり各エポックで以下を実行することである

1. Encoderで各動画のembeddingを取得する
2. Faissでembeddingの類似度でクラスタリングしてindexを設定する（代表点の設定）
3. ランダムな動画Vに対して2k個の近傍動画からk個の動画をランダムにサンプリングする
(直接k個の最近傍動画をサンプリングしないのは、最初に選んだ動画Vに近づけたいのではなく、クラスタ内の動画が互いに近くなるようにするため)

これを行うことで、k個の動画の各video-textクリップは正例、ある動画のvideo/textクリップと他の動画(Video1から見たVideo2など)のvideo/textクリップは負例(ハードネガティブ)となる。
よって、負例の数は正例の数よりもはるかに大きくなる（はず）。
![image](https://github.com/hypknot74/Paper-summaries/assets/57225520/e9f8de4f-e686-4f1f-b927-f8f93852120b)
